src/phoenix/server/api/types/Project.py

```diff
@@ -146,10 +146,22 @@ async def latency_ms_quantile(
         )
 
     @strawberry.field
-    def trace(self, trace_id: ID) -> Optional[Trace]:
-        if self.project.has_trace(TraceID(trace_id)):
-            return Trace(trace_id=trace_id, project=self.project)
-        return None
+    async def trace(self, trace_id: ID, info: Info[Context, None]) -> Optional[Trace]:
+        async with info.context.db() as session:
+            if (
+                trace_rowid := await session.scalar(
+                    select(models.Trace.id)
+                    .join(models.Project)
```

Comment:
When #2839 lands, we should add `project_rowid` as a private field on the `Project` type so we don't have to do this join.

Reply:
Created issue #2868

Reply:
Resolved in this PR